### PR TITLE
nix: rewrite nix flake from ground up and introduce home-manager module

### PR DESCRIPTION
**Pull request by NotAShelf**
Simplifies and cleans up the nix flake for further Nix development, also lays out groundwork for upcoming nixos or/and home-manager modules for anyrun.

### CODE REVIEW

File: .envrc
Comment (Kirottu) [inline .envrc:1]: If this file isn't required I'd rather not have it in the repo.

Comment (NotAShelf) [reply .envrc:1]: It is not required per se, but it helps hacking easier for direnv users.

Comment (Kirottu) [reply .envrc:1]: Yeah I'd prefer to not have it in the repo then.

Comment (NotAShelf) [reply .envrc:1]: It's not really possible to have it elsewhere (that has to be in the repo to trigger the direnv shell), but understood. I'll remove it in the next commit.

**Comment by n3oney** [comment]
home-manager module is here, and I think it’s pretty good
now @NotAShelf you could write the NixOS module based on it I guess, but do we really need one? I think we’ll be fine without one.
Something I think would be more important is making every plugin from this repository available as a separate package.

**Review by n3oney — APPROVED** [review]

**Comment by yavko** [comment]
Is there any help needed? I have experience writing nix flakes

**Comment by n3oney** [comment]
I think we can revert [`5965d84` (#44)](https://github.com/Kirottu/anyrun/pull/44/commits/5965d84cace6bdd57c33d2563f3f6bd1ff70994d), it's fine how it is now.
Still thinking about whether it needs a NixOS module though. Opinion? @NotAShelf

**Comment by NotAShelf** [comment]
I don't think a nixos module is necessary.

**Comment by NotAShelf** [comment]
I would prefer packaging plugins indicidually though, it seems like a major convenience to me, and a better approach than dabbling with fetchers for each plugin.

**Comment by n3oney** [comment]
pls remove the unneccessary buildInputs from the plugins :pray: https://github.com/n3oney/anyrun-ha-assist/blob/master/nix/default.nix

**Comment by n3oney** [comment]
lgtm mark it as ready @NotAShelf

**Comment by n3oney** [comment]
:smile:

**Pull request merged**